src/_pytest/python.py

```diff
@@ -684,6 +684,10 @@ def _get_first_non_fixture_func(obj, names):
 class Class(PyCollector):
     """ Collector for test methods. """
 
+    @classmethod
+    def from_parent(cls, parent, *, name, obj=None):
```

Comment:
Do you plan to introduce alternative constructors (other `classmethods`)? It would be nice to get those inplace already so we can see the benefits right away.

Reply:
at the moment, all i want is introduce the new utility so i can start to detangle __init__ from all the strange magic after deprecating/disallowing direct construction

other methods aren't on the radar, because atm thats not even the structural issue i`m trying to solve

the current problem i`m trying to solve is deep coupling in the node structures that's preventing stuff like `FunctionDefinition` introduction and `Instance` removal

Reply:
I see.

Do you think it is possible to introduce the new classmethods while deprecating `__init__`, so we can release this on `5.X`?

Reply:
yes actually

Reply:
Sounds good then!

I'm also interested to see how `Session.from_parent` (if it will exist) will look like.

Reply:
possibly not, session will likely have-to be split into multiple bits